### PR TITLE
Add type hints to toilStatus.py

### PR DESCRIPTION
**Pull request by jonathanxu18**
Adds type hints to toilStatus.py so it can be checked under mypy during linting.

May try to add more files with added type hints to this PR.

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/cliOptions.rst`
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [x] Make sure the PR passes tests.
* [x] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

File: src/toil/utils/toilStatus.py [hunk 1]
```diff
@@ -94,7 +96,7 @@ def printAggregateJobStats(self, properties, childNumber):
                              lf("READY_TO_RUN"), lf("IS_ZOMBIE"),
                              lf("HAS_SERVICES"), lf("IS_SERVICE"))))
 
-    def report_on_jobs(self):
+    def report_on_jobs(self) -> Dict[str, Any]:
```
Comment (jonathanxu18) [inline src/toil/utils/toilStatus.py:99]: I was trying to use Dict[str, List[Job]] as the type but mypy was throwing a warning unless Any was used. I'm going to try seeing if I can make the typing more specific.

Comment (DailyDreaming) [reply src/toil/utils/toilStatus.py:99]: Sounds good!

Comment (adamnovak) [reply src/toil/utils/toilStatus.py:99]: Some of the dict values are `List[Job]`, but `'properties'` is `Set[str]`, so the whole thing comes out to I think `Dict[str, Union[List[Job], Set[str]]]`, which isn't necessarily much more useful than `Dict[str, Any]` was.

**Review by adamnovak — APPROVED** [review]
I think this would be fine as is, although @jonathanxu18 if you are working on a more specific type for `report_on_jobs` we could wait for it.

**Comment by jonathanxu18** [comment]
> I think this would be fine as is, although @jonathanxu18 if you are working on a more specific type for `report_on_jobs` we could wait for it.

I think it probably wouldn't be much more useful to create a new type as you mentioned so I think we can just merge as is.

**Review by DailyDreaming — APPROVED** [review]
LGTM.

**Pull request merged**